### PR TITLE
(CLOUD-271) Added a simple example using a defined type

### PR DESCRIPTION
**Pull request by garethr**

### CODE REVIEW

File: examples/create-your-own-abstractions/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Create your own abstractions
+
+When describing any complex domain like your infrastructure it's
+important that you can introduce your own higher level concepts to talk
+about. In this examle we group together a number of AWS resources and
```
Comment (DavidS) [inline examples/create-your-own-abstractions/README.md]: s/examle/example

File: examples/create-your-own-abstractions/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+allow you to manage them together.
+
+
+## What
+
+In this example we'll setup 3 separate groups, each with 2 instances and 1
+security group. Because each group is nearly the same (apart from the
+name) we'll create our own type in Puppet, and then apply that 3 times.
+
+This is a good example of how to use Puppet to avoid repetetive effort
```
Comment (DavidS) [inline examples/create-your-own-abstractions/README.md]: s/repetetive/repetitive

File: examples/create-your-own-abstractions/somegroup.pp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  ec2_securitygroup { "${title}-sg":
+    ensure      => $ensure,
+    description => "group for ${title}",
+    region      => $region,
+  }
+
+  # ec2_instance resources need to be deleted before the
+  # associated security groups
+  if $ensure == 'absent' {
```
Comment (DavidS) [inline examples/create-your-own-abstractions/somegroup.pp]: I assume in the ensure => present case, it is the other way round and the the type does the correct auto require? In example code, this should be explained.

**Comment by DavidS** [comment]
Some typos and a request for clarification. Otherwise :+1:

**Comment by garethr** [comment]
@DavidS thanks, updates made and I've rebased against master so it now merges cleanly.

**Pull request merged**